packages/http/src/validator/index.ts

```diff
@@ -2,88 +2,61 @@ import { IPrismDiagnostic, IValidator } from '@stoplight/prism-core';
-        } else if (body) {
-          this.bodyValidator
-            .validate(body, (request && request.body && request.body.contents) || [], mediaType)
-            .forEach(validationResult => results.push(validationResult));
-        }
+    const { body } = input;
+    if (request && request.body) {
+      if (!body && request.body.required) {
+        results.push({ code: 'required', message: 'Body parameter is required', severity: DiagnosticSeverity.Error });
+      } else if (body) {
+        this.bodyValidator
+          .validate(body, (request && request.body && request.body.contents) || [], mediaType)
+          .forEach(validationResult => results.push(validationResult));
       }
+    }
 
-      this.headersValidator
-        .validate(input.headers || {}, (request && request.headers) || [], mediaType)
-        .forEach(validationResult => results.push(validationResult));
+    this.headersValidator
+      .validate(input.headers || {}, (request && request.headers) || [], mediaType)
+      .forEach(validationResult => results.push(validationResult));
 
-      this.queryValidator
-        .validate(input.url.query || {}, (request && request.query) || [], mediaType)
-        .forEach(validationResult => results.push(validationResult));
+    this.queryValidator
+      .validate(input.url.query || {}, (request && request.query) || [], mediaType)
+      .forEach(validationResult => results.push(validationResult));
 
-      return results;
-    }
-    return [];
+    return results;
   }
 
-  public validateOutput({
-    resource,
-    output,
-    config,
-  }: {
-    resource: IHttpOperation;
-    output?: IHttpResponse;
-    config?: IHttpConfig;
-  }): IPrismDiagnostic[] {
-    if (!output) {
-      return [];
-    }
-
-    if (!config || (config && config.validateResponse)) {
-      const results: IPrismDiagnostic[] = [];
-      const mediaType = caseless(output.headers || {}).get('content-type');
-      const responseSpec = resource.responses && findOperationResponse(resource.responses, output.statusCode);
-
-      this.bodyValidator
-        .validate(output.body, (responseSpec && responseSpec.contents) || [], mediaType)
-        .forEach(validationResult => results.push(validationResult));
+  public validateOutput({ resource, output }: { resource: IHttpOperation; output: IHttpResponse }): IPrismDiagnostic[] {
+    const results: IPrismDiagnostic[] = [];
+    const mediaType = caseless(output.headers || {}).get('content-type');
+    const responseSpec = resource.responses && findOperationResponse(resource.responses, output.statusCode);
 
-      this.headersValidator
-        .validate(output.headers || {}, (responseSpec && responseSpec.headers) || [], mediaType)
-        .forEach(validationResult => results.push(validationResult));
+    this.bodyValidator
+      .validate(output.body, (responseSpec && responseSpec.contents) || [], mediaType)
+      .forEach(validationResult => results.push(validationResult));
```

Comment:
any chance for a map/reduce instead of `forEach` and getting rid of `.push`ing?

Reply:
Yes, but it's not worth it since this is going dismantled very soon for a more generic approach. Spending time here is indeed worthless right now.

As next step when we'll be writing the contract testing piece we'll be refactoring most of this code.